libcxx/include/ccomplex

```diff
@@ -23,4 +23,17 @@
 #  pragma GCC system_header
 #endif
 
+#if _LIBCPP_STD_VER >= 17
+#  if _LIBCPP_STD_VER >= 20
+#    define _DEPRECATION_ANNOTATION _LIBCPP_DEPRECATED_("removed in C++20")
+#  else
+#    define _DEPRECATION_ANNOTATION _LIBCPP_DEPRECATED
+#  endif
+
+using __standard_header_ccomplex _DEPRECATION_ANNOTATION = void;
+using __use_standard_header_ccomplex                     = __standard_header_ccomplex;
+
+#  undef _DEPRECATION_ANNOTATION
+#endif
```

Comment:
This seems like a very complex way of writing
```c++
#if defined(__DEPRECATED) && __DEPRECATED
#  warning "<ccomplex> is deprecated"
#endif
```

Reply:
We're only doing this for `<hash_map>` & `<hash_set>`. I _guess_ the deprecation warnings should be controllable by `_LIBCPP_DISABLE_DEPRECATION_WARNINGS` (which is mentioned in `__config`).

Reply:
Yeah, we should probably slap a `&& !defined(_LIBCPP_DISABLE_DEPRECATION_WARNINGS)` at the end.

Reply:
I think that works as CI is green now.